### PR TITLE
Mixed code data model null pointer exception

### PR DESCRIPTION
**Pull request by blindmatrix**
MixedCodeDataModel threw a NullPointerException when hovering over references because some fields wasn't copied in the copy constructor.

Also marked read only fields as readonly.

### CODE REVIEW

File: src/UserInterfaces/WindowsForms/Controls/MixedCodeDataModel.cs [hunk 1]
```diff
@@ -164,9 +167,9 @@ private static Address Align(Address addr, uint alignment)
         /// Preemptively collect the machine code instructions
         /// in all image map blocks.
         /// </summary>
-        private void CollectInstructions()
+        private Dictionary<ImageMapBlock, MachineInstruction[]> CollectInstructions()
         {
-            this.instructions = new Dictionary<ImageMapBlock, MachineInstruction[]>();
+            Dictionary<ImageMapBlock, MachineInstruction[]> instructions = new Dictionary<ImageMapBlock, MachineInstruction[]>();
             foreach (var bi in imageMap.Items.Values.OfType<ImageMapBlock>()
```
Comment (uxmal) [inline src/UserInterfaces/WindowsForms/Controls/MixedCodeDataModel.cs:173]: Style: Prefer using `var instructions....` when creating an object with `new`, especially if the type of the created object is long and awkward.

**Review by uxmal — CHANGES_REQUESTED** [review]
I had a minor stylistic quibble, otherwise: nice job hunting down this issue!

File: src/UserInterfaces/WindowsForms/Controls/MixedCodeDataModel.cs (hunk 1 above)
Comment (blindmatrix) [reply src/UserInterfaces/WindowsForms/Controls/MixedCodeDataModel.cs:173]: Good catch, I also prefer var ..., Copy-paste mistake I guess :P

**Comment by uxmal** [comment]
Thanks for the contribution!

**Pull request merged**